whitesource/filters.py

```diff
@@ -0,0 +1,160 @@
+    for c in components:
+        artifacts_pre = c.sources if filter_cfg.type is whitesource.model.FilterType.SOURCE \
+            else c.resources
+        artifacts_past = list(filter(lambda a: _artifact_filter(
+            filter_cfg=filter_cfg,
+            artifact=a,
+        ), artifacts_pre))
+        if filter_cfg.type is whitesource.model.FilterType.SOURCE:
+            c.sources = artifacts_past
+        elif filter_cfg.type is whitesource.model.FilterType.RESOURCE:
+            c.resources = artifacts_past
+        else:
+            raise NotImplementedError(filter_cfg.type)
+
+        yield c
+
+
+def _artifact_filter(
+    filter_cfg: whitesource.model.WhiteSourceFilterCfg,
+    artifact: typing.Union[cm.ComponentSource, cm.Resource],
+):
+    if isinstance(filter_cfg.match, bool):
+        return _match_bool_filter(filter_cfg=filter_cfg, artifact=artifact)
+
+    elif isinstance(filter_cfg.match, dict):
+        # enum to string
+        if isinstance(artifact.type, cm.ResourceType):
+            resource_type_name = artifact.type.value
+        elif isinstance(artifact.type, str):
+            resource_type_name = str(artifact.type)
+        else:
+            raise NotImplementedError
+
+        re_str = resource_type_name if filter_cfg.type is whitesource.model.FilterType.RESOURCE \
+            else artifact.name
+        re_pat = filter_cfg.match.get('type') if filter_cfg.type is \
+            whitesource.model.FilterType.RESOURCE else filter_cfg.match.get('name')
+
+        if filter_cfg.action is whitesource.model.ActionType.INCLUDE:
+            if re.search(pattern=re_pat, string=re_str):
+                logger.debug(
+                    f'included {artifact.name=}, '
+                    f'reason: {re_str=} matches {re_pat=} and {filter_cfg.action=}'
+                )
+                return True
+        elif filter_cfg.action is whitesource.model.ActionType.EXCLUDE:
+            if not re.search(pattern=re_pat, string=re_str):
+                logger.debug(
+                    f'included {artifact.name=}, reason: '
+                    f'{re_str=} does not match {re_pat=} and {filter_cfg.action=}'
+                )
+                return True
+        else:
+            raise NotImplementedError(filter_cfg.action)
+    else:
+        raise NotImplementedError(filter_cfg.match)
+
+    logger.debug(f'excluded {artifact.name=}')
+    return False
+
+
+def _match_bool_filter(
+    filter_cfg: whitesource.model.WhiteSourceFilterCfg,
+    artifact: typing.Union[cm.Resource, cm.ComponentSource] = None,
+    component: cm.Component = None,
+) -> bool:
+
+    if not artifact and not component:
+        raise NotImplementedError
+
+    if filter_cfg.action is whitesource.model.ActionType.INCLUDE:
+        if filter_cfg.match:
+            logger.debug(f'included {artifact.name if artifact else component.name},'
+                         f' reason: {filter_cfg.match=} and {filter_cfg.action=}')
+            return True
```

Comment:
```suggestion
        return filter_cfg.match
```
Not sure if the debug code is actually needed tbh

Reply:
I have the gut feeling this won't be the last time we're touching this module and since the filtering logic is not too trivial I think keeping the output may help.